### ISSUE TITLE
mmcv.Config.fromstring not working on Windows

### ISSUE THREAD

**Issue by ypwhs** [open-mmlab/mmcv#1051]
**Describe the Issue**

Cannot load config on windows with `mmcv.Config.fromstring`. Because the file created by tempfile.NamedTemporaryFile cannot be opened twice on Windows.

**Reproduction**

```py
config = mmcv.Config.fromstring(checkpoint['meta']['config'], file_format='.py')
```

**Environment**

```
------------------------------------------------------------
sys.platform: win32
Python: 3.8.5 (default, Sep  3 2020, 21:29:08) [MSC v.1916 64 bit (AMD64)]
CUDA available: True
GPU 0: NVIDIA GeForce RTX 3090
CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.1
NVCC: Not Available
GCC: n/a
PyTorch: 1.8.1+cu111
PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 192829913
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 2019
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=C:/w/b/windows/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -openmp:experimental -DNDEBUG -DUSE_FBGEMM -DUSE_XNNPACK, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, 

TorchVision: 0.9.1+cu111
OpenCV: 4.5.1
MMCV: 1.3.4
MMCV Compiler: MSVC 191627045
MMCV CUDA Compiler: 11.1
MMDetection: 2.12.0+
------------------------------------------------------------
```

**Error traceback**

```
    config = mmcv.Config.fromstring(checkpoint['meta']['config'], file_format='.py')
  File "C:\Miniconda3\Lib\site-packages\mmcv\utils\config.py", line 282, in fromstring
    cfg = Config.fromfile(temp_file.name)
  File "C:\Miniconda3\Lib\site-packages\mmcv\utils\config.py", line 254, in fromfile
    cfg_dict, cfg_text = Config._file2dict(filename,
  File "C:\Miniconda3\Lib\site-packages\mmcv\utils\config.py", line 139, in _file2dict
    Config._substitute_predefined_vars(filename,
  File "C:\Miniconda3\Lib\site-packages\mmcv\utils\config.py", line 113, in _substitute_predefined_vars
    with open(filename, 'r', encoding='utf-8') as f:
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\ypw\\AppData\\Local\\Temp\\tmp3wmyrh7i.py'
```

**Bug fix**

Code: https://github.com/open-mmlab/mmcv/blob/58a84833520b3b5c71896ae45b9fa09831bd363d/mmcv/utils/config.py#L276-L279

> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later).

See: https://docs.python.org/zh-cn/3/library/tempfile.html

Solution: 

* create a temp file in `tempfile.TemporaryDirectory`.
* use `tempfile.mkstemp('.py')` to get a temp file path and then delete it

![image](https://user-images.githubusercontent.com/10473170/119607332-b636a000-be26-11eb-96cc-379d7316b2d1.png)

**Comment by zhouzaida** [comment]
hi @ypwhs , thanks for your feedback and solution, are you willing to create a pr?

**Comment by zhouzaida** [comment]
Resolved by #1077
